src/App.tsx

```diff
@@ -1,73 +1,111 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, []);
+
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const filteredPersonList = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 2000), []);
```

Comment:
```suggestion
  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
```
<img width="538" alt="image" src="https://github.com/mate-academy/react_autocomplete/assets/70220667/7c51cbe7-4d13-49e9-b32c-d3f260a798e3">